CONTRIBUTING.rst

```diff
@@ -27,13 +27,36 @@ Repository as a Pull Request.
 Pull Request Process
 --------------------
 
-1. Update the README.md with details of changes to the interface,
-   including new environment variables, proper file
-   locations, and container parameters.
-2. You may merge the Pull Request in once you have the sign-off of two
-   other developers, or if you do not have permission to do that, you
-   may request the second reviewer to merge it for you. We expect to
-   have a minimum of one approval from someone else on the core team.
+#. Fork RecTools `main repository <https://github.com/MobileTeleSystems/RecTools>`_
+   on GitHub. See `this guide <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo>`_ if you have questions.
+#. Clone your fork from GitHub to your local disk.
+#. Create a virtual environment and install dependencies including all 
+   extras and development dependencies.
+   
+   #. Make sure you have ``python3`` and ``poetry==1.4.0`` installed
+   #. Deactivate any active virtual environments. Deactivate conda ``base``
+      environment if applicable
+   #. Run ``make install`` command which will create a virtual env and
+      install everything that is needed with poetry. See `poetry usage details <https://python-poetry.org/docs/basic-usage/#installing-with-poetrylock>`_
+   
+#. Implement your changes. Check the following after you are done:
+   
+   #. Code styling. Autoformat with ``make format``
+   #. Linters. Run checks with ``make lint``
+   #. Tests. Make sure you've covered new features with tests. Check
+      code correctness with ``make test``
+   #. Coverage. Check with ``make coverage``
+   #. Docstrings. Please ensure that all new public classes and methods
```

Comment:
This point is covered with `make lint`.

We can remove it or put to the beginning of this list.

Reply:
moved to the beginning